pkg/operation/botanist/controlplane/kubescheduler/monitoring.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubescheduler
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const (
+	monitoringPrometheusJobName = "kube-scheduler"
+
+	monitoringMetricSchedulerBindingLatencyMicrosecondsBucket             = "scheduler_binding_latency_microseconds_bucket"
+	monitoringMetricSchedulerE2eSchedulingLatencyMicrosecondsBucket       = "scheduler_e2e_scheduling_latency_microseconds_bucket"
+	monitoringMetricSchedulerSchedulingAlgorithmLatencyMicrosecondsBucket = "scheduler_scheduling_algorithm_latency_microseconds_bucket"
+	monitoringMetricRestClientRequestsTotal                               = "rest_client_requests_total"
+	monitoringMetricProcessMaxFds                                         = "process_max_fds"
+	monitoringMetricProcessOpenFds                                        = "process_open_fds"
+
+	monitoringAlertingRules = `groups:
+- name: kube-scheduler.rules
+  rules:
+  - alert: KubeSchedulerDown
+    expr: absent(up{job="` + monitoringPrometheusJobName + `"} == 1)
+    for: 15m
+    labels:
+      service: ` + ServiceName + `
```

Comment:
Nit: I would rather use the deployment name for this label than the service name.
```suggestion
      service: ` + v1beta1constants.DeploymentNameKubeScheduler + `
```